src/core/blocks/create_block.c

```diff
@@ -49,7 +49,8 @@ unsigned int calc_tx_fees(Transaction *tx){
   unsigned long total_out = 0;
 
   for(unsigned int i = 0; i < tx->num_inputs; i++){
-    UTXO* input_utxo = utxo_pool_find(tx->inputs[i].prev_tx_id, tx->inputs[i].prev_utxo_output);
+    UTXO *input_utxo = NULL;// = utxo_pool_find(tx->inputs[i].prev_tx_id, tx->inputs[i].prev_utxo_output);
+    utxo_pool_find_leveldb(&input_utxo, tx->inputs[i].prev_tx_id, tx->inputs[i].prev_utxo_output);
```

Comment:
Needs a free

Reply:
We could also write find such that it takes an already allocated UTXO and uses the non-alloc deserialize to write into it... Would allow us to avoid having to catch all these frees where we don't actually need heap memory